modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/home/UserActivity.new.es.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayEmptyState from '@clayui/empty-state';
+import {useManualQuery} from 'graphql-hooks';
+import React, {useContext, useEffect, useState} from 'react';
+import {withRouter} from 'react-router-dom';
+
+import {AppContext} from '../../AppContext.es';
+import PaginatedList from '../../components/PaginatedList.es';
+import QuestionRow from '../../components/QuestionRow.es';
+import useQueryParams from '../../hooks/useQueryParams.es';
+import {getUserActivityQuery} from '../../utils/client.es';
+import {historyPushWithSlug} from '../../utils/utils.es';
+import {Question} from '../questions/Question.new.es';
+
+export default withRouter(
+	({
+		history,
+		location,
+		match: {
+			params: {creatorId},
+			url,
+		},
+	}) => {
+		const context = useContext(AppContext);
+		const queryParams = useQueryParams(location);
+		const siteKey = context.siteKey;
+		const [loading, setLoading] = useState(true);
+		const [page, setPage] = useState(null);
+		const [pageSize, setPageSize] = useState(null);
+		const [totalCount, setTotalCount] = useState(0);
+		const [currentQuestion, setCurrentQuestion] = useState({});
+
+		useEffect(() => {
+			const pageNumber = queryParams.get('page') || 1;
+			setPage(isNaN(pageNumber) ? 1 : parseInt(pageNumber, 10));
+		}, [queryParams]);
+
+		useEffect(() => {
+			setPageSize(queryParams.get('pagesize') || 20);
+		}, [queryParams]);
+
+		useEffect(() => {
+			document.title = creatorId;
+		}, [creatorId]);
+
+		const [fetchUserActivity, {data}] = useManualQuery(
+			getUserActivityQuery,
+			{
+				variables: {
+					filter: `creatorId eq ${creatorId}`,
+					page,
+					pageSize,
+					siteKey,
+				},
+			}
+		);
+
+		useEffect(() => {
+			if (!page || !pageSize) {
+				return;
+			}
+
+			setLoading(true);
+
+			fetchUserActivity().then(({data, loading}) => {
+				setTotalCount(data?.messageBoardMessages.totalCount || 0);
+				setLoading(loading);
+			});
```

Comment:
Loading is always false inside .then, so it's not necessary

```suggestion
			fetchUserActivity().then(({data}) => {
				setTotalCount(data?.messageBoardMessages.totalCount || 0);
				setLoading(false);
			});
```